gapic/src/main/com/google/gapid/perfetto/models/Tracks.java

```diff
@@ -55,14 +57,26 @@ private Tracks() {
   }
 
   private static ListenableFuture<Perfetto.Data.Builder> enumerateCpu(Perfetto.Data.Builder data) {
+    if (data.getNumCpus() == 0) {
+      return Futures.immediateFuture(data);
+    }
+
     CpuSummaryTrack summary = new CpuSummaryTrack(data.getNumCpus());
-    data.tracks.addLabelGroup(null, summary.getId(), "CPU Usage",
-        group(state -> new CpuSummaryPanel(state, summary), false, (group, filtered) -> {
-          for (int i = 1; i < group.size(); i += 2) {
-            group.setVisible(i, !filtered);
-          }
-        }, true));
-    return CpuTrack.enumerate(summary.getId(), data);
+    return transform(CpuTrack.enumerate(summary.getId(), data), configs -> {
+      boolean hasAnyFrequency = configs.stream().anyMatch(c -> c.hasFrequency);
+      Group.UiFactory ui = hasAnyFrequency ?
+          group(state -> new CpuSummaryPanel(state, summary), false, (group, filtered) -> {
+              for (int cpu = 0, track = 0; cpu < configs.size(); cpu++, track++) {
+                if (configs.get(cpu).hasFrequency) {
+                  track++;
```

Comment:
You're increasing ```track``` twice here. Is that the intended behavior?

Reply:
Ah, one track for the CPU, one track for the frequency, right?

Reply:
right.